src/main/kotlin/ecommerce/ProductController.kt

```diff
@@ -0,0 +1,58 @@
+package ecommerce
+
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseBody
+import java.net.URI
+
+@Controller
+class ProductController(private val productRepository: ProductRepository) {
```

Comment:
and u can also append this annotation 😉 
what is it's role?

```kotlin
@RequestMapping("/products")
```

and what's different between `@RestController` & `@Controller` ? 😉

Reply:
💯 U already know it well.

> shall I?

Yeah! for practice and grow 😄